src/layers/basic.jl

```diff
@@ -412,21 +432,32 @@ struct Parallel{F, T}
 end
 
 Parallel(connection, layers...) = Parallel(connection, layers)
+function Parallel(connection; kw...)
+  layers = NamedTuple(kw)
+  if :layers in Base.keys(layers) || :connection in Base.keys(layers)
+    throw(ArgumentError("a Parallel layer cannot have a named sub-layer called `connection` or `layers`"))
+  elseif isempty(layers)
+    Parallel(connection, ())
+  end
+  Parallel(connection, layers)
+end
 
 @functor Parallel
 
-(m::Parallel)(x::AbstractArray) = mapreduce(f -> f(x), m.connection, m.layers)
-(m::Parallel)(xs::Vararg{<:AbstractArray}) = mapreduce((f, x) -> f(x), m.connection, m.layers, xs)
+(m::Parallel)(x::AbstractArray) = mapreduce(f -> f(x), m.connection, Tuple(m.layers))
+(m::Parallel)(xs::Vararg{<:AbstractArray}) = mapreduce((f, x) -> f(x), m.connection, Tuple(m.layers), xs)
```

Comment:
Since we're now explicitly allowing empty branches.

```suggestion
(m::Parallel)(x::AbstractArray) = mapreduce(f -> f(x), m.connection, Tuple(m.layers); init = [])
(m::Parallel)(xs::Vararg{<:AbstractArray}) = mapreduce((f, x) -> f(x), m.connection, Tuple(m.layers), xs; init = [])
```

Reply:
Should this have a test?

Reply:
Probably should be `typeof(x)[]` but saving it for a separate PR is okay.

Reply:
In fact, is it obvious this should be `mapreduce`? The docstring doesn't really specify whether the combinator is applied pairwise of vararg. Its examples use `+`, where these are equivalent, ditto things like `Parallel(vcat, ...)`. But the vararg version would be more efficient (although I cannot imagine this being a bottleneck).

Reply:
It may break #1673 again

Reply:
I leave it alone for now? It's slightly odd to make an explicit branch for an empty cases which doesn't really work, but that case does mean that the PR overall doesn't change the behaviour here at all.

Reply:
I think we'd need to keep the explicit branch in anyways to not be breaking.

Reply:
At some point we should disallow it, #1673 is a good example where the right `init` would be hard to know.

Reply:
Yes that's what I mean. It's just that making it explicit makes it look more deliberate, whereas before it silently worked and you could imagine it was deliberate or an oversight.

Reply:
Well it was previously an accident. And I think it's clear now that we want to disallow it. I believe fixing an accidental oversight counts as a bug fix in semver and not a breaking change?

Reply:
Yes, I think disallowing it would be fine. I made #1685 now to discuss.